src/python/module/smartredis/srobject.py

```diff
@@ -0,0 +1,79 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2022, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from .smartredisPy import PySRObject
+from .util import exception_handler, typecheck
+
+from .error import *
+
+class SRObject:
+    def __init__(self, context):
+        """Initialize a SRObject object
+
+        :param context: logging context
+        :type name: str
+        """
+        typecheck(context, "context", str)
+        self._name = context
```

Comment:
Storing the ctx in a "protected" variable, but doesn't seem to be used internally within the class. Could be problematic if an app dev were to instance and then call `set_data`. Maybe remove or replace with
```py
@property
@exception_handler
def name(self):
    # Note: This would require adding new bindings
    self._data.get().get_context()
```

Reply:
Object-based context in Python isn't working yet, but will be resolved shortly as part of #338